File: ReverseInteger.py
'''
Given a signed 32-bit integer x, return x with its digits reversed. 
If reversing x causes the value to go outside the signed 32-bit integer range [-231, 231 - 1], then return 0.
'''
class Solution:
    def reverse(self, x: int) -> int:
        ans = int(str(abs(x))[::-1]) * (-1 if x < 0 else 1)
        return ans if -2**31 <= ans < 2**31 else 0

File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int revnum = 0;
        while (x != 0) {
            int rem = x % 10;
            x /= 10;
            if (revnum > Integer.MAX_VALUE / 10 || revnum < Integer.MIN_VALUE / 10) {
                return 0;
            }
            revnum = revnum * 10 + rem;
        }
        return revnum;
    }
}